tests/spirv/cbuffer-dx-layout-1.slang

```diff
@@ -0,0 +1,74 @@
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=CHECK):-vk -compute -xslang -fvk-use-dx-layout
+//TEST:SIMPLE(filecheck=SPIRV): -target spirv -entry computeMain -stage compute -fvk-use-dx-layout
+//TEST_INPUT:cbuffer(data=[1 2.0 3.0 4.0   5 6 7 0   8 9 10 11   12 0 0 0  13 0 0 0   14 0 0 0   15 0 0 0   16 17 18 0   19 20 21 22   23 24 0 0   25 0 0 0   26 0 0 0]):name=Test
+
+cbuffer Test
```

Comment:
Let's verify the layout with DXC's output to be sure it is correct.

Reply:
The output is the same, if it is preferred, I can add a `spirv-via-hlsl`/CROSS_SPIRV_HLSL test option so we can add a test using DXC directly.

Reply:
That isn't necessary. As long as it is verified to be correct it is all good.

Reply:
All layouts work as DXC does for SPIRV.
Attaching one of the 'framework' HLSL programs I used to test with for ease of future testing:
```
cbuffer Test
{
//SPIRV: Offset 0
    uint v0;
//SPIRV: Offset 4
    float3 v1;

//SPIRV: Offset 16
    uint3 v2;

//SPIRV: Offset 32
    uint2 v3;
//SPIRV: Offset 40
    uint2 v4;

//SPIRV: Offset 48
    uint v5[4];

// array always starts on a new register.
//SPIRV: Offset 112
    uint3 v6[2];
//SPIRV: Offset 140
// non-array can pack with a partially filled register
    uint v7;

//SPIRV: Offset 144
    uint2 v8;

// SPIRV: Offset 160
// array always starts on a new register.
    uint v9[2];
}

struct PSInput
{
	float4 color : COLOR;
};

float4 PSMain(PSInput input) : SV_TARGET
{
    float4 res;
    res.x+= v0;
    res.x+= v9[0];
	return res;
}
```